### DISCUSSION TITLE
Interrupts on ATtiny1604

### DISCUSSION THREAD

**Discussion by Raghu81** [SpenceKonde/megaTinyCore#634] [Q&A] [answered]
Referring to Attiny1604's datasheet, 
The interrupt vector table consists of only 2 vectors for PORTA and B respectively.. Which is common for both async and pin change interrupts..Therefore any interrupt on PORTA points to a common ISR... 
![IMG_20220127_195653](https://user-images.githubusercontent.com/71487301/151378589-9735b23f-3569-4c0b-8343-940d6e904fa6.jpg)

But when I used attachInterrupt() for 2 pins of PORTA ,where pin2 is a async interrupt and 3 is pinchange interrupt,  wrote seperate ISR for them ,it ran successfully and differentiated both ISR...  and interrupts on both pins executed different ISR's .....So how are interrupt request differentiated? Are they differentiated in attachInterrupt() function?? 
![INT1](https://user-images.githubusercontent.com/71487301/151380078-f9890adb-361e-4a87-83b8-1d191e5a5121.png)

If all the pins have seperate vectors.. How to refer to them?? 
And how to do it with ISR() function? What would be the vector for individual pins??

**Comment by SpenceKonde** [comment, marked as answer]
There is no vector for individual pins. 

attachInterrupt( keeps a table of up to 8 function pointers for each port, and the interrupt that fires loops up to 8 times checking whether that pin triggered an interrupt. That is why, even after my total refactor of attachInterrupt, which reduced the overhead considerably (compare the tools -> attachInterrupt mode default option with the old implementation, note the difference in binary size.), they are still grindingly slow. Not only is there that loop, but because a function that cannot be inlined (because it's set at runtime) is called from within an ISR (that's something you should never do), it has to push every call-used register onto the stack (16 of them), and then pop them back off at the end: 1 clock per push, and 2 per pop
